src/Sylius/Component/User/spec/Security/Generator/UniquePinGeneratorSpec.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Component\User\Security\Generator;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Component\User\Security\Checker\UniquenessCheckerInterface;
+use Sylius\Component\User\Security\Generator\GeneratorInterface;
+use Sylius\Component\User\Security\Generator\UniquePinGenerator;
+
+/**
+ * @mixin UniquePinGenerator
+ *
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+class UniquePinGeneratorSpec extends ObjectBehavior
+{
+    function it_throws_invalid_argument_exception_on_instantiation_with_non_integer_length(UniquenessCheckerInterface $checker)
+    {
+        $this->shouldThrow(\InvalidArgumentException::class)->during('__construct', [$checker, 'a string']);
+        $this->shouldThrow(\InvalidArgumentException::class)->during('__construct', [$checker, '8']);
+        $this->shouldThrow(\InvalidArgumentException::class)->during('__construct', [$checker, []]);
+        $this->shouldThrow(\InvalidArgumentException::class)->during('__construct', [$checker, new \StdClass()]);
+    }
+
+    function it_throws_invalid_argument_exception_on_instantiation_with_an_out_of_range_length(UniquenessCheckerInterface $checker)
```

Comment:
I'll move it below `let`, `it_is_initializable` and `it_implements_generator_interface`.